controllers/periodic_healthcheck_test.go

```diff
@@ -15,6 +21,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+func TestStartHealthCheckLoop(t *testing.T) {
+	g := NewWithT(t)
+	core, recordedLogs := observer.New(zapcore.InfoLevel)
+	logger := zapr.NewLogger(zap.New(core))
+
+	cluster := newClusterWithExternalEtcd()
+	etcdadmCluster := newEtcdadmCluster(cluster, withPausedAnnotation)
+	fakeClient := fake.NewClientBuilder().WithScheme(setupScheme()).WithObjects(etcdadmCluster).Build()
+
+	r := &EtcdadmClusterReconciler{
+		Client:              fakeClient,
+		Log:                 logger,
+		HealthCheckInterval: 1, // override the healthcheck interval to 1 second
+	}
+
+	done := make(chan struct{})
+
+	// Stop the healthcheck loop after 5 seconds
+	go func() {
+		time.Sleep(5 * time.Second)
+		close(done)
+	}()
```

Comment:
instead you could use t.Cleanup
It's simpler and you don't have to deal with concurrency

Reply:
removed